src/shared/components/palette/case-flag/write-case-flag-field.component.ts

```diff
@@ -171,20 +180,46 @@ export class WriteCaseFlagFieldComponent extends AbstractFieldWriteComponent imp
         this.errorMessages = component.errorMessages;
         this.formGroup.setErrors(component.errorMessages);
       } else {
-        // Populate new FlagDetail instance and add to the Flags data within the CaseField instance
-        if (this.fieldState === CaseFlagFieldState.FLAG_COMMENTS) {
-          const flagsCaseFieldValue = this.caseFlagParentFormGroup['caseField'].value;
-          flagsCaseFieldValue.details.push({value: this.populateNewFlagDetailInstance()});
+        switch (this.fieldState) {
+          case CaseFlagFieldState.FLAG_COMMENTS:
+            this.addFlagToCollection();
+            break;
+          case CaseFlagFieldState.FLAG_UPDATE:
+            this.updateFlagInCollection();
+            break;
+          default:
+            console.log('Invalid CaseFlagFieldState:', this.fieldState);
+            break;
         }
-        if (this.fieldState === CaseFlagFieldState.FLAG_UPDATE) {
-          // TODO: EUI-5342
-        }
-        // There is no error, update form group value and validity
-        this.formGroup.updateValueAndValidity();
       }
     }
   }
 
+  public addFlagToCollection(): void {
+    const flagsCaseFieldValue = this.caseFlagParentFormGroup['caseField'].value;
+    if (flagsCaseFieldValue) {
+      flagsCaseFieldValue.details.push({value: this.populateNewFlagDetailInstance()});
+      // There is no error, update form group value and validity
+      this.formGroup.updateValueAndValidity();
+    } else {
+      console.log('Case field value not found, flag cannot be added:', flagsCaseFieldValue);
+    }
+  }
+
+  public updateFlagInCollection(): void {
+    const flagsCaseFieldValue = this.caseFlagParentFormGroup['caseField'].value;
+    const flagDetailToUpdate = flagsCaseFieldValue.details.find(detail => detail.id === this.selectedFlagDetail.id);
+    if (flagDetailToUpdate) {
+      flagDetailToUpdate.value.flagComment = this.caseFlagParentFormGroup.value.flagComments
+        ? this.caseFlagParentFormGroup.value.flagComments
+        : null;
+      // There is no error, update form group value and validity
+      this.formGroup.updateValueAndValidity();
+    } else {
```

Comment:
@johnbenjamin-hmcts - not really sure how we should handle the `else` case. It shouldn't be the case that `flagDetailToUpdate` is falsy, yet if it is then is there anything the user can do? Maybe we should fail silently for now...

Reply:
@Daniel-Lam I have now updated the code to delete the else part.